src/rust/ide/src/controller/visualization.rs

```diff
@@ -0,0 +1,254 @@
+//! Visualization controller.
+//!
+//! Ths Visualization Controller is Responsible identifying all the available visualization natively
+//! embedded in IDE and available within the project's `visualization` folder. The
+//! `Visualization Controller` lives as long as the `Project Controller`.
+
+use crate::prelude::*;
+
+use std::rc::Rc;
+use enso_protocol::language_server;
+use graph_editor::component::visualization::class;
+use graph_editor::component::visualization::JsSourceClass;
+
+
+
+// =============
+// === Error ===
+// =============
+
+/// Enumeration of errors used in `Visualization Controller`.
+#[derive(Debug,Fail)]
+#[allow(missing_docs)]
+pub enum VisualizationError {
+    #[fail(display = "Visualization \"{}\" not found.", identifier)]
+    NotFound {
+        identifier : VisualizationIdentifier
+    },
+    #[fail(display = "JavaScript visualization \"{}\" failed to compile.", identifier)]
+    CompileError {
+        identifier : VisualizationIdentifier
+    }
+}
+
+
+
+// ===============================
+// === VisualizationIdentifier ===
+// ===============================
+
+/// This enum is used to identify a visualization both in the project folder or natively embedded in
+/// IDE.
+#[derive(Clone,Debug,Display,Eq,PartialEq)]
+#[allow(missing_docs)]
+pub enum VisualizationIdentifier {
```

Comment:
I don't like that we have both `VisualizationIdentifier` and `VisualizationId` and that they mean totally different things.

Reply:
@mwu-tow how do you feel about `VisualizationPath`?

Reply:
`VisualizationPath` is fine.